accel/tcg/cputlb.c

```diff
@@ -1559,8 +1598,21 @@ static int probe_access_internal(CPUArchState *env, target_ulong addr,
     tlb_addr = tlb_read_ofs(entry, elt_ofs);
 
     page_addr = addr & TARGET_PAGE_MASK;
-    if (!tlb_hit_page(tlb_addr, page_addr)) {
-        if (!victim_tlb_hit(env, mmu_idx, index, elt_ofs, page_addr)) {
+
+    bool tag_write_invalid = false;
+
+#ifdef TARGET_CHERI
+    if (access_type == MMU_DATA_CAP_STORE &&
+        (env_tlb(env)
+             ->d[mmu_idx]
+             .iotlb[tlb_index(env, mmu_idx, addr)]
+             .tagmem_write == TLBENTRYCAP_INVALID_WRITE))
```

Comment:
Surely this isn't the right formatting?..

Reply:
I agree it looks really weird, but that's what clang-format does. Not sure what QEMU style is here.

Reply:
I do what the mighty clang-format gods compel me to do.